gtk/gtkfilechooserwidget.c

```diff
@@ -6180,15 +6180,32 @@ gtk_file_chooser_widget_get_default_size (GtkFileChooserEmbed *chooser_embed,
   GtkFileChooserWidget *impl = GTK_FILE_CHOOSER_WIDGET (chooser_embed);
   GtkFileChooserWidgetPrivate *priv = impl->priv;
   GtkRequisition req;
-  int x, y, width, height;
+  int x, y, width, height, max_width, max_height;
   GSettings *settings;
+  GdkScreen *screen;
+  GdkWindow *window;
+  gint monitor_num;
+  GdkRectangle monitor;
+
+  screen = gtk_widget_get_screen (GTK_WIDGET (impl));
+  window = gtk_widget_get_window (GTK_WIDGET (impl));
```

Comment:
You could use gdk_screen_get_root_window() will wont fail if the widget is not realized yet